src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2304,6 +2304,110 @@ class InsertStatementSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class MergeStatementSegment(BaseSegment):
+    """A `MERGE` statement."""
+
+    type = "merge_statement"
+    match_grammar = StartsWith(
+        Sequence("MERGE", "INTO"),
+    )
+    parse_grammar = Sequence(
+        "MERGE",
+        "INTO",
+        OneOf(
+            Ref("TableReferenceSegment"),
+            Ref("AliasedTableReferenceGrammar"),
+        ),
+        "USING",
+        Indent,
+        OneOf(
+            Ref("TableReferenceSegment"),
+            Ref("AliasedTableReferenceGrammar"),
+            Sequence(
+                Bracketed(
+                    Ref("SelectableGrammar"),
+                ),
+                Ref("AliasExpressionSegment"),
+            ),
+        ),
+        Dedent,
+        Ref("JoinOnConditionSegment"),
+        Ref("MergeMatchSegment"),
+    )
+
+
+@ansi_dialect.segment()
+class MergeMatchSegment(BaseSegment):
+    """Contains dialect specific merge operations.
+
+    hookpoint for dialect specific behavior
+    e.g. UpdateClause / DeleteClause, multiple MergeMatchedClauses
+    """
+
+    type = "merge_match"
+    match_grammar = Sequence(
+        Ref("MergeMatchedClauseSegment"),
+        Ref("MergeNotMatchedClauseSegment"),
+    )
+
+
+@ansi_dialect.segment()
+class MergeMatchedClauseSegment(BaseSegment):
+    """The `WHEN MATCHED` clause within a `MERGE` statement."""
+
+    type = "merge_when_matched_clause"
+    match_grammar = Sequence(
+        "WHEN",
+        "MATCHED",
+        Indent,
+        "THEN",
+        Ref("MergeUpdateClauseSegment"),
+        Dedent,
+    )
+
+
+@ansi_dialect.segment()
+class MergeNotMatchedClauseSegment(BaseSegment):
+    """The `WHEN NOT MATCHED` clause within a `MERGE` statement."""
+
+    type = "merge_when_not_matched_clause"
+    match_grammar = Sequence(
+        "WHEN",
+        "NOT",
+        "MATCHED",
+        "THEN",
+        Indent,
+        Ref("MergeInsertClauseSegment"),
+        Dedent,
+    )
+
+
+@ansi_dialect.segment()
+class MergeUpdateClauseSegment(BaseSegment):
+    """`UPDATE` clause within the `MERGE` statement."""
+
+    type = "merge_update_clause"
+    match_grammar = Sequence(
+        "UPDATE",
+        Ref("SetClauseListSegment"),
+    )
+
+
+@ansi_dialect.segment()
+class MergeInsertClauseSegment(BaseSegment):
+    """`INSERT` clause within the `MERGE` statement."""
+
+    type = "merge_insert_clause"
+    match_grammar = Sequence(
+        "INSERT",
+        Indent,
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        Dedent,
+        Ref("ValuesClauseSegment", optional=True),
```

Comment:
Should this be wrapped in another `Indent`/`Dedent`?

```suggestion
        Indent,
        Ref("ValuesClauseSegment", optional=True),
        Dedent,
```

Reply:
agree, done